pki.spec

```diff
@@ -1111,7 +1120,31 @@ then
     cp /usr/share/java/resteasy/resteasy-jackson2-provider.jar \
         resteasy-jackson2-provider-$RESTEASY_VERSION.jar
 
-    ls -l
+    #migrate necessary files being copied around to jakarta 9.0 ee
+
+    /usr/bin/javax2jakarta -logLevel=ALL -profile=EE  jboss-jaxrs-api_2.0_spec-$JAXRS_VERSION.jar  jboss-jaxrs-api_2.0_spec-$JAXRS_VERSION.jar 
+
+    /usr/bin/javax2jakarta -logLevel=ALL -profile=EE  jackson-jaxrs-json-provider-$JACKSON_VERSION.jar jackson-jaxrs-json-provider-$JACKSON_VERSION.jar
+
+    /usr/bin/javax2jakarta -logLevel=ALL -profile=EE  jackson-annotations-$JACKSON_VERSION.jar jackson-annotations-$JACKSON_VERSION.jar
+    /usr/bin/javax2jakarta -logLevel=ALL -profile=EE  jackson-core-$JACKSON_VERSION.jar  jackson-core-$JACKSON_VERSION.jar
+    /usr/bin/javax2jakarta -logLevel=ALL -profile=EE  jackson-databind-$JACKSON_VERSION.jar jackson-databind-$JACKSON_VERSION.jar
+    /usr/bin/javax2jakarta -logLevel=ALL -profile=EE  jackson-jaxrs-base-$JACKSON_VERSION.jar  jackson-jaxrs-base-$JACKSON_VERSION.jar
+    /usr/bin/javax2jakarta -logLevel=ALL -profile=EE  jackson-jaxrs-json-provider-$JACKSON_VERSION.jar jackson-jaxrs-json-provider-$JACKSON_VERSION.jar 
+    /usr/bin/javax2jakarta -logLevel=ALL -profile=EE  jackson-module-jaxb-annotations-$JACKSON_VERSION.jar jackson-module-jaxb-annotations-$JACKSON_VERSION.jar
+
+    /usr/bin/javax2jakarta -logLevel=ALL -profile=EE   jakarta.activation-api-$JAKARTA_ACTIVATION_API_VERSION.jar  jakarta.activation-api-$JAKARTA_ACTIVATION_API_VERSION.jar
+    /usr/bin/javax2jakarta -logLevel=ALL -profile=EE   jakarta.annotation-api-$JAKARTA_ANNOTATION_API_VERSION.jar jakarta.annotation-api-$JAKARTA_ANNOTATION_API_VERSION.jar  
+    /usr/bin/javax2jakarta -logLevel=ALL -profile=EE   jakarta.xml.bind-api-$JAXB_API_VERSION.jar jakarta.xml.bind-api-$JAXB_API_VERSION.jar 
+
+    # Add local artifact so we can compile against the migrated jboss-jaxrs-api_2.0_spec-$JAXRS_VERSION.jar
+    # We could have used the maven install plugin but it's not available with standard rpms.
+
+    # Create the local artifact structure
+    mkdir -p ~/.m2/repository/pki-local/jboss-jaxrs-api_2.0_spec/$JAXRS_VERSION
+    # Copy over the jaxrs api so we can compile
+    cp jboss-jaxrs-api_2.0_spec-$JAXRS_VERSION.jar  ~/.m2/repository/pki-local/jboss-jaxrs-api_2.0_spec/$JAXRS_VERSION/jboss-jaxrs-api_2.0_spec-$JAXRS_VERSION.jar
```

Comment:
We need to see if these commands are allowed in Fedora build system.

Reply:
Yeah, I can explain the need for all of this. The compilation takes place in a few phases. It seems like the first phase depends on the pom / artifacts declared. I needed to create a local artifact for the api file. I think in subsequent phases our stuff takes over and stuff is compiled against the contents of common/lib, which has all the stuff I've migrated. I attempted to use the install plugin but fedora doesn't have that enabled, so I looked up how to do it kind of manually. But yes, that could be an issue.